hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -243,6 +243,10 @@ private OMResponse submitRequestToRatis(OMRequest request)
 
   private OMResponse submitReadRequestToOM(OMRequest request)
       throws ServiceException {
+    // Read from leader or followers using linearizable read
+    if (omRatisServer.isLinearizableRead()) {
+      return handler.handleReadRequest(request);
+    }
```

Comment:
> ... OzoneManagerRequestHandler#handleReadRequest queries the OM metadata tables directly without going through Ratis (OzoneManagerStateMachine#query). ...

This is a good point!  Adding the code here seems incorrect.

> ... does OM provides a "read-after-write" consistency (or is it stronger?) even through the read does not go through Ratis server? 

For the same client, since the calls are blocking, it will have "read-after-write" consistency (An earlier write must be done before processing the later read.)

> ... If we enable linearizable read we can increase the consistency guarantee to "linearizable"?

Yes.

It also guarantee read-after-write consistent even for the Ratis async APIs.

Reply:
Thank you for the info.

> Adding the code here seems incorrect.

I think this can be fixed by also calling `submitRequestToRatis` even on the read path.

I'm curious, is there any historical reason that currently OM does not send read request to OM Ratis server, maybe some overhead consideration? My understanding is that it should not have a large additional overhead since read request through Ratis (`StateMachine#query`) does not generate an append log entry.

Reply:
> ...  currently OM does not send read request to OM Ratis server?  ...

For non-ha, there is no Ratis server in OM.  It probably is the reason.